packages/block-library/src/image/image.js

```diff
@@ -499,43 +499,55 @@ export default function Image( {
 		[ clientId, isSingleSelected, metadata?.bindings ]
 	);
 
+	const shouldRenderOtherControls =
+		( isSingleSelected &&
+			! isEditingImage &&
+			! lockHrefControls &&
+			! lockUrlControls ) ||
+		allowCrop ||
+		( isSingleSelected && canInsertCover );
+
 	const controls = (
 		<>
-			<BlockControls group="block">
-				{ isSingleSelected &&
-					! isEditingImage &&
-					! lockHrefControls &&
-					! lockUrlControls && (
-						<ImageURLInputUI
-							url={ href || '' }
-							onChangeUrl={ onSetHref }
-							linkDestination={ linkDestination }
-							mediaUrl={ ( image && image.source_url ) || url }
-							mediaLink={ image && image.link }
-							linkTarget={ linkTarget }
-							linkClass={ linkClass }
-							rel={ rel }
-							showLightboxSetting={ showLightboxSetting }
-							lightboxEnabled={ lightboxChecked }
-							onSetLightbox={ onSetLightbox }
-							resetLightbox={ resetLightbox }
+			{ shouldRenderOtherControls && (
```

Comment:
A better name might be `showBlockControls` because they belong to the `block` group. The `other` group is below in the component tree. 

Also, for consistency, we use `show` instead of `render` verb in variable names here to indicate that something is being shown (rendered).